x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/get_filter.ts

```diff
@@ -107,6 +107,9 @@ export const getFilter = async ({
         throw new BadRequestError('savedId parameter should be defined');
       }
     }
+    case 'machine_learning': {
+      throw new Error('getFilter called with a ML Rule');
```

Comment:
I think you want like you have above which is a `BadRequestError`. Something like this:

```ts
throw new BadRequestError('getFilter cannot be called with a ML Rule');
```

Reply:
This code is not used for ML rules; this was to satisfy the linter complaining about the `assertUnreachable`. Given that, what would you suggest?

Reply:
I'd say we use the `BadRequestError` as if we ever have a bug where it suddenly becomes possible for some very rare or add reason this makes it through we will get the BadRequest being thrown.